### PR TITLE
docs: add top level commands to help text, restructure

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

Add the new top level commands from [the ongoing refactoring](https://github.com/wasmCloud/wash/issues/538) to the Wasmcloud help text and restructure it.

Help text now looks like this:

```
_________________________________________________________________________________
                               _____ _                 _    _____ _          _ _
                              / ____| |               | |  / ____| |        | | |
 __      ____ _ ___ _ __ ___ | |    | | ___  _   _  __| | | (___ | |__   ___| | |
 \ \ /\ / / _` / __| '_ ` _ \| |    | |/ _ \| | | |/ _` |  \___ \| '_ \ / _ \ | |
  \ V  V / (_| \__ \ | | | | | |____| | (_) | |_| | (_| |  ____) | | | |  __/ | |
   \_/\_/ \__,_|___/_| |_| |_|\_____|_|\___/ \__,_|\__,_| |_____/|_| |_|\___|_|_|
_________________________________________________________________________________

Interact and manage wasmCloud applications, projects, and runtime environments

Usage: wash [OPTIONS] <COMMAND>

Configure:
  completions  Generate shell completions for wash
  ctx          Manage wasmCloud host configuration contexts
  drain        Manage contents of local wasmCloud caches
  keys         Utilities for generating and managing keys
  claims       Generate and manage JWTs for wasmCloud actors

Build:
  new          Create a new project from template
  build        Build (and sign) a wasmCloud actor, capability provider, or interface
  inspect      Inspect capability provider or actor module
  par          Create, inspect, and modify capability provider archive files

Run:
  up           Bootstrap a local wasmCloud environment
  down         Tear down a local wasmCloud environment (launched with wash up)
  app          Manage declarative applications and deployments (wadm) (experimental)

Iterate:
  start        Start an actor or provider
  link         Link an actor and a provider
  call         Invoke a wasmCloud actor
  stop         Stop an actor or provider, or host
  ctl          Interact with a wasmCloud control interface

Publish:
  pull         Pull an artifact from an OCI compliant registry
  push         Push an artifact to an OCI compliant registry
  reg          Perform operations on an OCI or Bindle registry

Optimize:
  gen          Generate code from smithy IDL files
  lint         Perform lint checks on smithy models
  validate     Perform validation checks on smithy models

Options:
  -o, --output <OUTPUT>  Specify output format (text or json) [default: text]
  -h, --help             Print help
  -V, --version          Print version
```

## Related Issues

#538 

## Release Information

`next`

## Consumer Impact

Consumers will have a more intuitive help text that maps to the process of using wasmCloud

## Testing

Built on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### CODE REVIEW

**Comment by vados-cosmonic** [comment]
Note: @brooksmtownsend had [some feedback](https://wasmcloud.slack.com/archives/C04RVMJJH17/p1686145764082899?thread_ts=1686136199.258239&cid=C04RVMJJH17):

- [x] refactor ordering

File: src/main.rs [hunk 1]
```diff
@@ -66,33 +66,42 @@ Interact and manage wasmCloud applications, projects, and runtime environments
...
   inspect      Inspect capability provider or actor module
-  lint         Perform lint checks on smithy models
-  new          Create a new project from template
   par          Create, inspect, and modify capability provider archive files
-  reg          Push an actor or provider component to an OCI or Bindle registry
+
+Run:
+  up           Bootstrap a local wasmCloud environment
+  down         Tear down a local wasmCloud environment (launched with wash up)
+  app          Manage declarative applications and deployments (wadm) (experimental)
```
Comment (connorsmith256) [inline src/main.rs]: Are we ready to remove the experimental tag from the wadm commands yet?

**Review by connorsmith256 — DISMISSED** [review]

File: src/main.rs (hunk 1 above)
Comment (brooksmtownsend) [reply src/main.rs]: We really should, that's a way old tag

Comment (brooksmtownsend) [inline src/main.rs]:
```suggestion
  app          Manage declarative applications and deployments (wadm)
```

**Review by brooksmtownsend — DISMISSED** [review]

File: src/main.rs (hunk 1 above)
Comment (vados-cosmonic) [reply src/main.rs]: done, thank you!

Comment (vados-cosmonic) [reply src/main.rs]: :green_circle:  Done, thanks for making the commit suggestion!

**Review by brooksmtownsend — APPROVED** [review]
🚀

**Pull request merged**